### PR TITLE
friendly message when channel is null

### PR DESCRIPTION
**Pull request by SergeyKanzhelev**
fixing #45

### CODE REVIEW

File: src/Core/Managed/Shared/TelemetryClient.cs [hunk 1]
```diff
@@ -334,6 +334,11 @@ public void Track(ITelemetry telemetry)
 
                 telemetry.Sanitize();
 
+                if (this.Channel == null)
```
Comment (abaranch) [inline src/Core/Managed/Shared/TelemetryClient.cs]: Why this is not the first check we do after IsEnabled?

File: src/Core/Managed/Shared/TelemetryClient.cs [hunk 2]
```diff
@@ -360,7 +365,7 @@ public void Initialize(ITelemetry telemetry)
             var telemetryWithProperties = telemetry as ISupportProperties;
             if (telemetryWithProperties != null)
             {
-                if (this.Channel.DeveloperMode.HasValue && this.Channel.DeveloperMode.Value)
+                if ((this.Channel != null) && (this.Channel.DeveloperMode.HasValue && this.Channel.DeveloperMode.Value))
```
Comment (abaranch) [inline src/Core/Managed/Shared/TelemetryClient.cs]: Why would we try to enrich data that we will drop? Let's remove this check and throw before we get here

**Comment by SergeyKanzhelev** [comment]
The reason doing so is to enable VS integration in future when we will debug output it before throwing an exception.

**Comment by abaranch** [comment]
:shipit:

**Pull request merged**